soc/arm/nxp_kinetis/k2x/soc.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2019 bytes at work AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief System/hardware module for k2x platform
+ *
+ * This module provides a generic routine to setup a k2x based board. It expects
+ * a clock_config.c generated by 'MCUXpresso Configuration Tools'
+ * (https://mcuxpresso.nxp.com/en/clock) in the 'boards' folder.
+ */
+
+#include <init.h>
+
+#if defined(CONFIG_SOC_MK22FA12)
+#include <system_MK22FA12.h>
+#else
+#error "Unknown Soc"
+#endif
+
+static int system_wrapper(struct device *arg)
```

Comment:
If you look at the K6X soc.c, the clocks are setup based on Kconfig options here. You have DTS clock divider settings for the SOC above, probably doable with those defines